nixos/modules/services/networking/squid.nix

```diff
@@ -95,6 +95,10 @@ let
     refresh_pattern .               0       20%     4320
   '');
 
+  squidConfig = pkgs.runCommand "squid-validated.conf" {} ''
+    ${pkgs.squid}/bin/squid -k parse -f "${squidConfigRaw}"
```

Comment:
This will also fail when you use certificates:

```
squid-validated.conf> 2021/08/01 08:20:15| ERROR: unable to load certificate file '/var/lib/acme/thalheim.io/fullchain.pem': error:02001002:system library:fopen:No such file or directory
squid-validated.conf> 2021/08/01 08:20:15| WARNING: 'HTTPS_port [::]:8889' missing certificate in '/var/lib/acme/thalheim.io/fullchain.pem'
```

Reply:
I see the problem. (annoying that "parse" actually means more than parsing) :)

In that case, I think I'll prefer my alternate suggestion, i.e. expose an option for "configFile", which defaults to writeText of configText.

This would at least make it easier for me to generate (and optionally verify) the config file myself.

Reply:
Another possible approach (which depends on squid error messages) is to only fail the test if `squid -k` prints the tested filename on stdout. This way it means that it complains on something that was wrong with the configuration file itself instead of something missing in the sandbox.